# HG changeset patch
# User Andreas Regel <andreas.regel@gmx.de>
# Date 1303056014 -7200
# Node ID 76ce46a10b7686ea46674905d4e08283db83b1f3
# Parent 2195a7d275aa3ecc468de263301a928e31a64b54
locally define DVB OSD API extensions

Allow compiling the plugin even when using DVB header files that don't contain the needed definitions for the S2-6400 command interface.

--- a/hdffcmd.c	Sun Apr 17 13:20:22 2011 +0200
+++ b/hdffcmd.c	Sun Apr 17 18:00:14 2011 +0200
@@ -12,6 +12,25 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <vdr/tools.h>
+
+#if !defined OSD_RAW_CMD
+typedef struct osd_raw_cmd_s {
+    const void *cmd_data;
+    int cmd_len;
+    void *result_data;
+    int result_len;
+} osd_raw_cmd_t;
+
+typedef struct osd_raw_data_s {
+    const void *data_buffer;
+    int data_length;
+    int data_handle;
+} osd_raw_data_t;
+
+#define OSD_RAW_CMD            _IOWR('o', 162, osd_raw_cmd_t)
+#define OSD_RAW_DATA           _IOWR('o', 163, osd_raw_data_t)
+#endif
+
 
 namespace HDFF
 {

